### PR TITLE
Make new_saturating methods public

### PR DESCRIPTION
**Pull request by V02460**
I also added doc comments with examples ~~and shortened the implementation~~.

### CODE REVIEW

**Comment by V02460** [comment]
Didn’t know I used a new and fancy feature :smile:

File: src/uint.rs [hunk 1]
```diff
@@ -90,9 +90,20 @@ impl UInt {
         Self(val & MAX_SAFE_UINT)
     }
 
-    // TODO: make public if name is deemed sensible, rename and make public otherwise.
+    /// Creates an `UInt` from the given `i64` capped at `MAX_SAFE_UINT`.
```
Comment (jplatte) [inline src/uint.rs:93]:
```suggestion
    /// Creates an `UInt` from the given `u64` capped at `MAX_SAFE_UINT`.
```

**Comment by jplatte** [comment]
What new and fancy feature did you use? `.clamp`?

**Comment by V02460** [comment]
Haha, yes :) Good to have the CI reminding one that library backwards compatibility is a thing.

**Pull request merged**